render/render_test.go

```diff
@@ -510,3 +512,46 @@ func TestRenderReaderNoContentLength(t *testing.T) {
 	assert.Equal(t, headers["Content-Disposition"], w.Header().Get("Content-Disposition"))
 	assert.Equal(t, headers["x-request-id"], w.Header().Get("x-request-id"))
 }
+
+func TestRenderJsonPb(t *testing.T) {
+	w := httptest.NewRecorder()
+	reps := []int64{int64(1), int64(2)}
+	typ := int32(11)
+	label := "test"
+	data := &testdata.Test{
+		Label:         &label,
+		Type:          &typ,
+		Reps:          reps,
+		Optionalgroup: &testdata.Test_OptionalGroup{RequiredField: &label},
+	}
+
+	r := (JSONPB{
+		Data:   data,
+		datapb: []byte{},
+		Option: protojson.MarshalOptions{
+			EmitUnpopulated: true,
+		},
+	})
+	r.WriteContentType(w)
+	result, err := protojson.MarshalOptions{EmitUnpopulated: true}.Marshal(data)
+	assert.NoError(t, err)
+	err = r.Render(w)
+	assert.NoError(t, err)
+	assert.Equal(t, string(result), w.Body.String())
+	assert.Equal(t, "application/json; charset=utf-8", w.Header().Get("Content-Type"))
+
+	w1 := httptest.NewRecorder()
+	r1 := (JSONPB{
+		Data:   nil,
+		datapb: []byte{},
+		Option: protojson.MarshalOptions{
+			EmitUnpopulated: true,
+		},
+	})
+	r1.WriteContentType(w)
```

Comment:
Wondering why using the previous httptest.NewRecorder `w`?
Could this be a typo? Or maybe I miss something?

Moreover, maybe testing JSONPB with `nil` data (`r1`) can be separated into another test?